docs/DeveloperGuide.md

```diff
@@ -1,38 +1,331 @@
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
+## Setup & Prerequisites
+1. Ensure you have `Java 11` installed.
+2. Ensure your local repository is synced with the main repository at [AY2223S2-CS2113-F13-1/tp](https://github.com/AY2223S2-CS2113-F13-1/tp)
+3. Download the latest `tp.main.jar` from [here](https://github.com/AY2223S2-CS2113-F13-1/tp/releases/tag/v1.0).
+4. Copy the file to the folder you want to use as home folder for the recipe manager.
+5. Use `Win+R` to open the command prompt and type `cmd` and press Enter.
+6. Then `cd` into the folder where you copied the jar file. e.g. `cd C:\Users\Lee\Desktop\MyRecipe`
+7. Type `java -jar tp.main.jar` and press Enter to start the program.
+
 ## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+### Architecture
+
+**Main components of the architecture**
+
+![image](./PlantUML/MainArchitecture.png)
+
+The ***Architecture Diagram*** given above explains the high-level design of the App.
+
+Given below is a quick overview of Duke(main) components and how they interact with each other.
+It is responsible for,
+* At app launch: Initializes all other components in the correct sequence, and connects them up with each other.
+* At shut down: Shuts down the components and invokes cleanup methods where necessary.
+
+The rest of the App consists of five components.
+
+* [**`UI`**](#ui-component): The UI of the App.
+* [**`Parser`**](#parser-component): Parses user input.
+* [**`Command`**](#command-component): The command executor.
+* [**`RecipeList`**](#recipelist-component): Holds the recipe data in the recipe manager by an ArrayList. 
+* [**`Storage`**](#storage-component): Reads data from, and writes data to, the hard disk.
+
+**How the architecture components interact with each other**
+The *Sequence Diagram* below shows how the components interact with each other for the scenario where the user issues the command `delete 1`.
+![image](./PlantUML/ArchitectureInteract.png)
+
+#### UI component
+The **API** of this component is specified in [`UI.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/ui/UI.java)
+
+<!-- ![image](./PlantUML/UIcomponent.png) -->
+
+UI class implements the StringLib interface for some output strings. It is responsible for the following tasks:
+* Prints the welcome and goodbye messages.
+* Prints the help message, which lists all the commands available and their examples.
+* Prints the log messages for some managing operations.
+* Prints the error messages for some exceptions.
+
+#### Parser component
+The **API** of this component is specified in [`Parser.java`](https://github.com/AY2223S2-CS2113-F13-1/tp/blob/master/src/main/java/seedu/duke/parser/Parser.java)
+
+![image](./PlantUML/ParserComponent.png)
+
```

Comment:
Remove the C and I symbols beside the Class and Interface boxes as it is not the module convention. Replace the private and public access modifiers with - and + symbols instead.